src/Juvix/FrontendContextualise/ModuleOpen/Environment.hs

```diff
@@ -0,0 +1,96 @@
+{-# LANGUAGE LiberalTypeSynonyms #-}
+
+module Juvix.FrontendContextualise.ModuleOpen.Environment
+  ( module Juvix.FrontendContextualise.ModuleOpen.Environment,
+    module Juvix.FrontendContextualise.Environment,
+  )
+where
+
+import qualified Juvix.Core.Common.Context as Context
+import Juvix.FrontendContextualise.Environment
+import qualified Juvix.FrontendContextualise.ModuleOpen.Types as New
+import qualified Juvix.FrontendDesugar.RemoveDo.Types as Old
+import Juvix.Library
+import qualified Juvix.Library.HashMap as Map
+
+type Old f =
+  f (NonEmpty (Old.FunctionLike Old.Expression)) Old.Signature Old.Type
+
+type New f =
+  f (NonEmpty (New.FunctionLike New.Expression)) New.Signature New.Type
+
+type WorkingMaps m =
+  ( HasState "old" (Old Context.T) m,
+    HasState "new" (New Context.T) m,
+    HasThrow "error" Error m,
+    HasState "modMap" ModuleMap m
```

Comment:
Do you think it's better to alias `HasState "modMap" ModuleMap m` separately so we can use it below?

Reply:
Since we don't use this signature much, it's fine to stay how it is.

Reply:
Alright!